### PR TITLE
Added algorithm for Sum-tree

### LINKED ISSUE #1365: [Idea]: <Sum Tree>Sum Tree
### Idea Title

Sum Tree

### Name:  
```[NEW ALGORITHM]```  Given a binary tree, the task is to check if it is a Sum Tree. A Sum Tree is a Binary Tree where the value of a node is equal to the sum of the nodes present in its left subtree and right subtree. An empty tree is Sum Tree and the sum of an empty tree can be considered as 0. A leaf node is also considered a Sum Tree.

---
### About:  
Implementation of binary tree

### Input:
The nodes of the tree.

### Output:
True or False ( Based on calculating the value of child nodes and parent node)

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shuvojitss**
### Description
Added the sum tree program under tree .Given a binary tree, the task is to check if it is a Sum Tree. A Sum Tree is a Binary Tree where the value of a node is equal to the sum of the nodes present in its left subtree and right subtree. An empty tree is Sum Tree and the sum of an empty tree can be considered as 0. A leaf node is also considered a Sum Tree .This program is w.r.t issue #1365 which is based on additon of this algoriithm.

Fixes #1365

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Trees/Sum-tree.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+### Sample Input:
+    //       26
+    //      /  \
+    //     10   3
+    //    / \    \
+    //   4  6     3           
+
+### Sample Output:
+    The binary tree is a Sum Tree.
+
```
Comment (ajay-dhangar) [inline docs/Trees/Sum-tree.md]: Rewrite in good format

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/Trees/Sum-tree.md (hunk 1 above)
Comment (shuvojitss) [reply docs/Trees/Sum-tree.md]: Should I remove the comments in input. I have just described the tree structurally. Can you please tell me what changes you want.??

**Review by shuvojitss — COMMENTED** [review]
@ajay-dhangar Changed the input section.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**